pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -122,7 +123,16 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 // ValidateInfrastructureConfigUpdate validates a InfrastructureConfig object.
 func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *apisgcp.InfrastructureConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newConfig.Networks, oldConfig.Networks, fldPath.Child("networks"))...)
+
+	vpcPath := field.NewPath("networks.vpc")
+
+	oldVPC := oldConfig.Networks.VPC
+	newVPC := newConfig.Networks.VPC
+
+	if oldVPC != nil && newVPC != nil {
+		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newVPC.Name, oldVPC.Name, vpcPath.Child("name"))...)
+		allErrs = append(allErrs, apivalidation.ValidateImmutableField(newVPC.CloudRouter, oldVPC.CloudRouter, vpcPath.Child("cloudRouter"))...)
+	}
```

Comment:
We also should prevent changing [those fields](https://github.com/gardener/gardener-extension-provider-gcp/blob/master/pkg/apis/gcp/v1alpha1/types_infrastructure.go#L40-L47), or?

Reply:
What about the internal CIDR? Shouldn't it be immutable as well?